### ISSUE TITLE
Typemap looking when using template and typedef

### ISSUE THREAD

**Issue by diorcety** [swig/swig#55]
Imagine you put a typemap like this:
`typemap(out) Namespace::A "Example";`
Added to this Namespace::A is a typedef to Namespace::B

```
namespace Namespace {
typedef B A;
}
```

now use a vector of Namespace::A
`%template(Namespace::A) std::vector<Namespace::A>;`

There is an problem! For example in python for getitem function it will try the following typemap:
`std::vector< Namespace::B >::value_type`
then

```
Namespace::B
```

A correct behaviour should be

```
std::vector< Namespace::A >::value_type
Namespace::A
std::vector< Namespace::B >::value_type
Namespace::B
```

or

```
std::vector< Namespace::A >::value_type
std::vector< Namespace::B >::value_type
Namespace::A
Namespace::B
```

**Comment by ojwb** [comment]
It looks like this may have been addressed by #112 and that this issue was overlooked at the time, but I don't follow exactly how to reproduce this issue to check.

@diorcety Can you retest and close if this is already fixed?

**Comment by ojwb** [comment]
No response from the submitter for nearly 18 months, but the testcase in the associated PR#54 passes with git master, so I'm closing this now.
